modules/kubernetes-addons/agones/main.tf

```diff
@@ -1,26 +1,33 @@
 module "helm_addon" {
-  source            = "../helm-addon"
+  source = "../helm-addon"
+
   manage_via_gitops = var.manage_via_gitops
-  helm_config       = local.helm_config
-  irsa_config       = null
-  addon_context     = var.addon_context
 
-  depends_on = [kubernetes_namespace_v1.this]
-}
+  helm_config = merge(
+    {
+      name             = "agones"
+      chart            = "agones"
+      repository       = "https://agones.dev/chart/stable"
+      version          = "1.23.0"
+      namespace        = "agones-system"
+      create_namespace = true
+      description      = "Agones Gaming Server Helm Chart deployment configuration"
+      values           = [file("${path.module}/values.yaml")]
+    },
+    var.helm_config
+  )
```

Comment:
@bobdoah Any reason why you are refactoring all the add-ons?

Reply:
@vara-bonthu  I made most of the changes. When reviewing the PR, there were too many inconsistencies across the different addons in how names, nampespaces, etc. + the merging of provided config with `var.helm_config` that I just dove in to start correcting. Please let me know how you would like to handle these changes so that the users can opt in or out of creating namespaces while keeping the addons consistent